robotoff/elasticsearch/export.py

```diff
@@ -0,0 +1,55 @@
+import orjson
+from elasticsearch import Elasticsearch
+
+from robotoff import settings
+from robotoff.elasticsearch.category.dump import generate_category_data
+from robotoff.elasticsearch.product.dump import generate_product_data
+from robotoff.utils import get_logger
+from robotoff.utils.es import perform_export
+
+logger = get_logger()
+
+
+class ElasticsearchExporter:
+    """ElasticsearchExporter exports new index data to Elasticsearch."""
+
+    def __init__(self, es_client: Elasticsearch):
+        self.es_client = es_client
+
+    def _delete_existing_data(self, index: str) -> None:
+        resp = self.es_client.delete_by_query(
```

Comment:
I think request arriving while cleaning index should be 503 (service unavailable), as this is the case here, service is not available.

Here this means, some requests will be returned as 200 while matched against a half empty index ? (delete is asynchronous in ES and you don't have any notion of transaction).

I'm not sure, but I feel it's better to fail with 503 for the time of reindexation.

Note1: If you want to handle this with no down time, a technique is to use an alias instead of index name, then create the new index, reroute the alias to it, and remove the old index (but this is far fetched for now, and this would need a new PR indeed).

Note2: also here you run a delete_by_query but it may end before deleting all data (there is a timeout), not sure how it will react then (does it raise, but in this case we will have an empty index for whole day ?).

Reply:
Yes, let's open a separate issue for this